packages/automerge-repo-network-messagechannel/src/index.ts

```diff
@@ -118,3 +111,18 @@ interface MessageChannelNetworkAdapterConfig {
    */
   useWeakRef?: boolean
 }
+
+//types
+
+type ArriveMessage = {
```

Comment:
duplicated types here

Reply:
This is part of the reason that i felt these messages could be brought into Repo and out of the adapters. Technically `join` and `peer` on the websocket connections are doing exactly the same thing.

Reply:
@pvh not sure where might the best place for these types? There is actually a lot of duplicated code across the broadcastchannel and messagechannel adapters, which could easily be refactored if there were a sensible place to put the common code?